x-pack/solutions/security/plugins/elastic_assistant/server/ai_assistant_data_clients/knowledge_base/helpers.ts

```diff
@@ -157,7 +157,11 @@ export const getStructuredToolForIndexEntry = ({
   }, {});
 
   return new DynamicStructuredTool({
-    name: indexEntry.name.replace(/[^a-zA-Z0-9-]/g, ''), // // Tool names expects a string that matches the pattern '^[a-zA-Z0-9-]+$'
+    name: indexEntry.name
+      // Ensure it starts with a letter. If not, prepend 'a'
+      .replace(/^[^a-zA-Z]/, 'a')
+      // Replace invalid characters with an empty string
+      .replace(/[^a-zA-Z0-9_]/g, ''),
```

Comment:
Flipping these around would mean this `-testing` would become `testing` instead of `atesting`.
```suggestion
      // Replace invalid characters with an empty string
      .replace(/[^a-zA-Z0-9_]/g, '')
      // Ensure it starts with a letter. If not, prepend 'a'
      .replace(/^[^a-zA-Z]/, 'a'),
```

Reply:
good idea, ill do it